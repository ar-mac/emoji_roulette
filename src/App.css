.App {
  text-align: center;
}

body {
  font-size: 30px;
}

button {
  font-size: 20px;
}

input {
  font-size: 30px;
}

img {
  height: 30px;
}
