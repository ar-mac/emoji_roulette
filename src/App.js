import React, { Component } from 'react';

import Roulette from './components/Roulette/Roulette';
import RegisterForm from './components/RegisterForm/RegisterForm';
import './App.css';

class App extends Component {
  state = {
    isRegistered: false,
  };

  setRegistrationStatus = (isRegistered) => {
    this.setState({ isRegistered })
  };

  render() {
    const { isRegistered } = this.state;
    return (
      <div className="App">
        <h2>Welcome to Emoji Routlette</h2>
        {
          isRegistered
            ? <Roulette />
            : <RegisterForm handleSubmit={this.setRegistrationStatus} />
        }
      </div>
    );
  }
}

export default App;
