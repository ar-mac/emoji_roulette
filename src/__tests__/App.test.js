import React from 'react';
import App from '../App';

describe('<App />', () => {
  const defaultProps = {};

  const setup = buildSetup(App, defaultProps, { lifecycleExperimental: true });

  test('renders', () => {
    expect(setup().wrapper.instance()).toMatchSnapshot();
  });

  test('renders roulette component when isRegistered', () => {
    const { wrapper } = setup();
    wrapper.setState({ isRegistered: true });
    expect(wrapper.instance()).toMatchSnapshot();
  });

  test('setRegistrationStatus sets isRegistered state to provided value', () => {
    const { wrapper } = setup();
    wrapper.instance().setRegistrationStatus(true);
    expect(wrapper.state('isRegistered')).toEqual(true);
  });
});
