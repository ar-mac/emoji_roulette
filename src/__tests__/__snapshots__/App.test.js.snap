// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
App {
  "_reactInternalInstance": ShallowComponentWrapper {
    "_calledComponentWillUnmount": false,
    "_compositeType": 0,
    "_context": Object {},
    "_currentElement": <App />,
    "_debugID": 1,
    "_hostContainerInfo": null,
    "_hostParent": null,
    "_instance": [Circular],
    "_mountOrder": 1,
    "_pendingCallbacks": null,
    "_pendingElement": null,
    "_pendingForceUpdate": false,
    "_pendingReplaceState": false,
    "_pendingStateQueue": null,
    "_renderedComponent": NoopInternalComponent {
      "_currentElement": <div
        className="App container"
>
        <h2>
                Welcome to Super Emoji Routlette
        </h2>
        <withBackend(Component)
                handleSubmit={[Function]}
        />
</div>,
      "_debugID": 2,
      "_renderedOutput": <div
        className="App container"
>
        <h2>
                Welcome to Super Emoji Routlette
        </h2>
        <withBackend(Component)
                handleSubmit={[Function]}
        />
</div>,
    },
    "_renderedNodeType": 0,
    "_rootNodeID": 0,
    "_topLevelWrapper": null,
    "_updateBatchNumber": null,
    "_warnedAboutRefsInRender": false,
  },
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "setRegistrationStatus": [Function],
  "state": Object {
    "isRegistered": false,
  },
  "updater": Object {
    "enqueueCallback": [Function],
    "enqueueCallbackInternal": [Function],
    "enqueueElementInternal": [Function],
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
    "validateCallback": [Function],
  },
}
`;

exports[`<App /> renders roulette component when isRegistered 1`] = `
App {
  "_reactInternalInstance": ShallowComponentWrapper {
    "_calledComponentWillUnmount": false,
    "_compositeType": 0,
    "_context": Object {},
    "_currentElement": <App />,
    "_debugID": 3,
    "_hostContainerInfo": null,
    "_hostParent": null,
    "_instance": [Circular],
    "_mountOrder": 2,
    "_pendingCallbacks": null,
    "_pendingElement": null,
    "_pendingForceUpdate": false,
    "_pendingReplaceState": false,
    "_pendingStateQueue": null,
    "_renderedComponent": NoopInternalComponent {
      "_currentElement": <div
        className="App container"
>
        <h2>
                Welcome to Super Emoji Routlette
        </h2>
        <Roulette />
</div>,
      "_debugID": 4,
      "_renderedOutput": <div
        className="App container"
>
        <h2>
                Welcome to Super Emoji Routlette
        </h2>
        <Roulette />
</div>,
    },
    "_renderedNodeType": 0,
    "_rootNodeID": 0,
    "_topLevelWrapper": null,
    "_updateBatchNumber": null,
    "_warnedAboutRefsInRender": false,
  },
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "setRegistrationStatus": [Function],
  "state": Object {
    "isRegistered": true,
  },
  "updater": Object {
    "enqueueCallback": [Function],
    "enqueueCallbackInternal": [Function],
    "enqueueElementInternal": [Function],
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
    "validateCallback": [Function],
  },
}
`;
